File: Programmers/java/LEVEL_1_추억_점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        int idx = 0;

        Map<String, Integer> map = new HashMap<>();

        for (int i = 0; i < name.length; i++) {
            map.put(name[i], yearning[i]);
        }

        for (String[] p : photo) {
            int num = 0;

            for (String n : p) {
                if (map.containsKey(n)) {
                    num += map.get(n);
                }
            }

            answer[idx++] = num;
        }

        return answer;
    }
}

File: Programmers/cpp/LEVEL_1_추억_점수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
	vector<int> answer;
	unordered_map<string, int> um;

	for (int i = 0; i < name.size(); i++) {
		um[name[i]] = yearning[i];
	}

	for (vector<string> p : photo) {
		int num = 0;

		for (string name : p) {
			if (um.count(name) != 0) {
				num += um[name];
			}
		}

		answer.push_back(num);
	}

	return answer;
}